sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1338,17 +1338,36 @@ export class ApplicationService extends RecordDataModelService<Application> {
       .getOne();
   }
 
-  async doesApplicationExist(
-    applicationNumber: string,
-    studentId: number,
-  ): Promise<boolean> {
-    return !!(await this.repo.findOne({
+  /**
+   * Check if a Student Application exists.
+   * @param options query options.
+   * - `applicationId` application id.
+   * - `applicationNumber` application number.
+   * - `studentId` student id, useful for authorization to ensure
+   * the student has access to the application.
+   * @returns true if the application exists, otherwise false.
+   */
+  async doesApplicationExist(options: {
+    applicationId?: number;
+    applicationNumber?: string;
+    studentId?: number;
+  }): Promise<boolean> {
+    if (
+      !options.applicationId &&
+      !options.applicationNumber &&
+      !options.studentId
```

Comment:
If only studentId is passed to this method, the behavior would not be as expected and the validation here will not catch that. I would suggest to use overload like this to enforce the validation at compile time. 

```ts
async doesApplicationExist(options: {
    applicationId: number;
    applicationNumber?: string;
    studentId?: number;
  }): Promise<boolean>;

  async doesApplicationExist(options: {
    applicationId?: number;
    applicationNumber: string;
    studentId?: number;
  }): Promise<boolean>;
```

Reply:
I did not consider it critical. It would be similar if it is a student API call and the student ID is not provided, which would not enforce the authorization, my point being that the method consumer can be responsible for providing the necessary parameters.
Please let me know it it is a suggestion or a blocker.

Reply:
We have a validation here(currently in the code) to validate that consumer is using it appropriately but on top of having the validation if consumer can miss authorization then I would consider it as blocker and ask for change.

Reply:
Regarding the "We have a validation here(currently in the code) to validate that consumer is using it appropriately", the assumption is not accurate and the validation in place is to ensure that the method is called with at least one parameter.
The intention of the method is to be generic and potentially be used by Students and by the Ministry where the student ID will be optional.

Reply:
Changed as per team discussion.